ViewEnvironment/Sources/EnvironmentValues+ViewEnvironment.swift

```diff
@@ -29,4 +29,28 @@ extension EnvironmentValues {
     }
 }
 
+
+extension Environment where Value == ViewEnvironment {
+    
+    @available(
+        *,
+         deprecated,
+         message:
+            """
+            Please do not create an `@Environment` property that references the top-level `viewEnvironment`: \
+            it will break SwiftUI's automatic invalidation when any part of the `ViewEnvironment` changes. \
+            Instead, reference your relevant sub-property, eg `@Environment(\\.viewEnvironment.myProperty)`.
+            """
+    )
+    @inlinable public init(_ keyPath: KeyPath<EnvironmentValues, Value>) {
+        fatalError(
```

Comment:
is fatal'ing unconditionally the right approach? what exactly is the downside if one does this? if it's 'just' poor performance, should we be more lenient with the production code and make this an assertion?

Reply:
I think a fatal is correct / preferred, since you should never do this in production code. If we already had consumers of SwiftUI doing this, I'd probably go w/ an assert, but given we have the opportunity to "close the door" on this permanently, I'd say let's do that